sql/src/main/java/org/apache/druid/sql/calcite/rel/Windowing.java

```diff
@@ -280,12 +289,21 @@ private static List<OperatorFactory> computeWindowOperations(
       if (!sortMatches(priorSortColumns, sortColumns)) {
         // Sort order needs to change. Resort and repartition.
         ops.add(new NaiveSortOperatorFactory(new ArrayList<>(sortColumns)));
-        ops.add(new NaivePartitioningOperatorFactory(group.getPartitionColumns()));
+        if (sqlEngine instanceof NativeSqlEngine) {
+          ops.add(new NaivePartitioningOperatorFactory(group.getPartitionColumns()));
+        } else {
+          ops.add(new GlueingPartitioningOperatorFactory(group.getPartitionColumns()));
+        }
```

Comment:
> MSQ translation layer part

Can you elaborate which layer do you mean by that?

Do you mean WindowOperatorQueryKit? I originally did that, but it fails the tests of MSQWindowTest because they also validate the generated MSQ spec, which still had NaivePartitioningOperatorFactory (as that happens prior to the query kit layer).

Reply:
Based on offline discussion, will pass plannerContext instead of engine to this method, and will create a separate private method for the duplicate if-else lines.

Reply:
This comment is not relevant anymore with the latest changes on the PR, hence closing it.